src/backend/distributed/master/citus_tools.sql

```diff
@@ -0,0 +1,204 @@
+/*
+ * citus_tools.sql
+ * Contains definitions of citus_tools UDFs
+ * - citus_run_on_all_workers
+ * - citus_run_on_all_placements
+ * - citus_run_on_all_colocated_placements
+ * - citus_run_on_shards
+ *
+ * These functions depends on presense of UDF master_run_on_worker
+ */
+
+CREATE OR REPLACE FUNCTION master_run_on_worker(worker_name text[], port integer[],
+												command text[], parallel boolean,
+												OUT node_name text, OUT node_port integer,
+												OUT success boolean, OUT result text )
+	RETURNS SETOF record
+	LANGUAGE C STABLE STRICT
+	AS 'citus.so', $function$;
+
+
+CREATE OR REPLACE FUNCTION citus_shard_name(table_name regclass, shard_id bigint)
```

Comment:
hold the review for C part guys.
I am brushing up helper UDF so that parallel flag would mean something.